tests/python_functional/src/helpers/loggen/conftest.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python
+#############################################################################
+# Copyright (c) 2020 One Identity
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+import pytest
+from pathlib2 import Path
+from psutil import TimeoutExpired
+
+import src.testcase_parameters.testcase_parameters as tc_parameters
+from src.executors.process_executor import ProcessExecutor
+from src.syslog_ng.syslog_ng_paths import SyslogNgPaths
+
+
+class Loggen(object):
+
+    def __init__(self, testcase_parameters):
+        self.loggen_proc = None
+        self.loggen_stdout_path = Path(tc_parameters.WORKING_DIR, "loggen_stdout")
+        self.loggen_stderr_path = Path(tc_parameters.WORKING_DIR, "loggen_stderr")
+        self.instance_path = SyslogNgPaths(testcase_parameters).set_syslog_ng_paths("server")
+        self.loggen_bin_path = self.instance_path.get_loggen_bin()
+
+    def _decode_start_parameters(
+        self, inet, unix, tcp, udp, use_ssl, dont_parse, read_file, skip_tokens, loop_reading,
+        rate, interval, permanent, syslog_proto, proxied, sdata, no_framing, active_connections,
+        idle_connections, ipv6, debug, number, csv, quiet,
+    ):
+
+        start_parameters = []
+
+        if inet is True:
+            start_parameters.append("--inet")
+
+        if unix is True:
+            start_parameters.append("--unix")
+
+        if tcp is True:
+            start_parameters.append("--stream")
+
+        if udp is True:
+            start_parameters.append("--dgram")
+
+        if use_ssl is True:
+            start_parameters.append("--use-ssl")
+
+        if dont_parse is True:
+            start_parameters.append("--dont-parse")
+
+        if read_file is True:
```

Comment:
This should be `is not None`, similarly to the one below.

Reply:
@furiel: I think we don't want to add the flag, if `read_file=False` is set. `is not None` is true for `False` I believe.

Reply:
`read_file` should not be boolean, it is a string (filename).

```
file-reader
  -R, --read-file=<filename>                 Read log messages from file
```

Reply:
Yes, you are right! My bad.